library/src/main/java/com/davidstemmer/screenplay/flow/Screenplay.java

```diff
@@ -1,145 +1,139 @@
 package com.davidstemmer.screenplay.flow;
 
+import android.app.Activity;
 import android.view.View;
 import android.view.ViewGroup;
 
-import com.davidstemmer.screenplay.SceneCut;
 import com.davidstemmer.screenplay.SceneState;
-import com.davidstemmer.screenplay.Stage;
 import com.davidstemmer.screenplay.scene.Scene;
+import com.davidstemmer.screenplay.util.CollectionUtils;
 
 import java.util.ArrayDeque;
+import java.util.Collection;
+import java.util.Deque;
 import java.util.Iterator;
 
-import flow.Backstack;
-import flow.Flow;
-
 
 /**
  * The Screenplay object handles the navigation logic for a Screenplay application.
  */
-public class Screenplay implements Flow.Listener {
+public class Screenplay {
 
     private SceneState screenState = SceneState.NORMAL;
 
-    private final Stage stage;
-
-    private final ArrayDeque<Scene> previousBackstack = new ArrayDeque<>();
+    private final Activity activity;
+    private final ViewGroup container;
 
-    public Screenplay(Stage stage) {
-        this.stage = stage;
+    public Screenplay(Activity activity, ViewGroup container) {
+        this.activity = activity;
+        this.container = container;
     }
 
-    @Override
-    public void go(Backstack backstack, Flow.Direction direction, Flow.Callback callback) {
+    public enum Direction {
```

Comment:
seriously though, I like enums. You can add add variables, constructors and methods to them which you can't do with ints. If use an int now later decide I need an enum, I'd have to change the public facing API.